### PR TITLE
Electron - open file in working directory when launching RStudio with a file arg

### PR DESCRIPTION
**Pull request by jgutman**
### Intent

Address #[#11778](https://github.com/rstudio/rstudio/issues/11778), at least on Windows and Linux. May need some additional handling for Mac to process double-click on Finder events, but will leave to subsequent PR.

### Approach

Set the RS_INITIAL_WD environment variable, used by the C++ session code, in pre-launch Electron code

### Automated Tests
None

### QA Notes
Will need to be tested on at least Windows and Mac -- see notes in issue.

### CODE REVIEW

File: src/node/desktop/src/main/application-launch.ts [hunk 1]
```diff
@@ -70,19 +71,18 @@ export class ApplicationLaunch {
...
     if (existsSync(projectFile)) {
-      setenv('RS_INITIAL_PROJECT', projectFile);
+      setenv(kRStudioInitialProject, projectFile);
     }
 
     // run it
     spawn(process.execPath, argv, { detached: true });
 
     // restore environment variables
-    unsetenv('RS_INSTALL_WD');
```
Comment (jgutman) [inline src/node/desktop/src/main/application-launch.ts:85]: RS_INSTALL_WD doesn't even exist, pretty sure that was a typo and RS_INITIAL_WD was intended. These get unset in the C++ code anyway, so I'm not sure that it matters

File: src/node/desktop/src/main/args-manager.ts [hunk 2]
```diff
@@ -146,8 +146,12 @@ export class ArgsManager {
             setenv(kRStudioInitialProject, arg);
             return false;
           }
+          else {
```
Comment (jgutman) [inline src/node/desktop/src/main/args-manager.ts:149]: for all files that are not Rproj files

Comment (timtmok) [reply src/node/desktop/src/main/args-manager.ts:149]: You might be able to do the same thing for the `open-file` event. However, the event may be called at any time and not necessarily before the session launches. The electron docs specifies that the `open-file` event at startup needs to be handled before the app is ready. But I'm not sure if that guarantees that opening from Finder will always fire the event before app ready.

I guess try that in the next PR and find out! 🤞

**Review by timtmok — APPROVED** [review]

File: src/node/desktop/src/main/args-manager.ts (hunk 2 above)
Comment (jgutman) [reply src/node/desktop/src/main/args-manager.ts:149]: yep, working on that PR now -- seems like we can just have an if/else clause based on whether or not the app is ready yet. That's what we do for handling the Rproj files so I am guessing the same thing will work for other files

**Pull request merged**